README.md

```diff
@@ -30,18 +30,108 @@ Input JSON file with road OpenStreetMap tags:
 }
 ```
 
-Output lane specifications from left to write:
+Output: A list of lanes from left to write according to the [specification](./data/spec-lanes.json).
+JSON formatted example:
 
 ```json
 [
-    {"type": "sidewalk", "direction": "backward"},
-    {"type": "cycleway", "direction": "forward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "backward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "travel_lane", "direction": "forward"},
     {"type": "sidewalk", "direction": "forward"}
 ]
 ```
 
+Lane Definition
+---------------
+
+It is important to determine a strict definition for a lane.
+An OpenStreetMap way, usually a `highway=*`, can in some cases be divided into lanes,
+where each lane extends from the start to the end of the way.
+A lane's direction of travel is not necessarily the same as the direction of the way.
+A lane represents a continuous area with no formal separation of vehicles.
+Lanes are often separated by something visible like painted lines or raised curbs,
+but these are not necessary for lanes to exist.
+
+
+### Example 1 ###
+
+A road with 5 marked lanes.
+
+```json
+[
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "forward"},
+]
+```
+
+![Example 1](https://upload.wikimedia.org/wikipedia/commons/thumb/0/0e/Gr%C3%BCnbergstra%C3%9Fe_2.JPG/240px-Gr%C3%BCnbergstra%C3%9Fe_2.JPG)
+
+Note: there is no sidewalk or verge for pedestrians to walk here.
+
+### Example 2 ###
+
+A road with 2 marked lanes.
+
+```json
+[
+    {"type": "shoulder", "direction": "both"},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "shoulder", "direction": "both"},
+]
+```
+
+![Example 2](https://upload.wikimedia.org/wikipedia/commons/f/f0/A537_Cat_and_Fiddle_Road_-_geograph.org.uk_-_175899.jpg)
+
+Note:
+
+- there is are shoulders where pedestrians could walk in both directions.
+- this is in the UK, so traffic drives on the left.
+- the direction is the predominant direction,
+  overtaking vehicles could travel in the opposite direction.
+
+### Example 3 ###
+
+A narrow road with one lane.
```

Comment:
I like modeling this as one travel lane with `direction=both`. I hereby regret my early choice of creating two skinny one-direction lanes to handle these.